File: 459b/main.cpp
#include <iostream>

using namespace std;

int flowers[200005];

int main() {
    int n;
    cin >> n;

    int min = 1000000005;
    int max = 0;
    long long minn = 0, maxn = 0;
    for (int i = 0; i < n; i++) {
        cin >> flowers[i];
        if (flowers[i] > max) {
            max = flowers[i];
        }
        if (flowers[i] < min) {
            min = flowers[i];
        }
    }

    int diff = max - min;
    for (int i = 0; i < n; i++) {
        if (flowers[i] == min) {
            minn += 1;
        } else if (flowers[i] == max) {
            maxn += 1;
        }
    }

    if (minn == n) {
        cout << 0 << " " << (long long)n * (n - 1) / 2;
    } else {
        cout << diff << " " << minn * maxn << endl;
    }
}

File: 459b/main.go
package main

import "fmt"

var flowers [200005]int

func main() {
	var n int
	fmt.Scan(&n)

	min := 10000000000
	max := 0
	minn, maxn := 0, 0
	for i := 0; i < n; i++ {
		fmt.Scan(&flowers[i])
		if flowers[i] > max {
			max = flowers[i]
		}

		if flowers[i] < min {
			min = flowers[i]
		}
	}

	diff := max - min
	for i := 0; i < n; i++ {
		if flowers[i] == min {
			minn += 1
		} else if flowers[i] == max {
			maxn += 1
		}
	}

	fmt.Println(diff, int64(minn*maxn))
}
